src/middleware.test.ts

```diff
@@ -0,0 +1,67 @@
+import { middleware } from './middleware'
+import { NextResponse } from 'next/server'
+import type { NextRequest } from 'next/server'
+import { beforeEach, describe, expect, it, vi } from 'vitest'
+
+// Mock NextResponse methods
+vi.mock('next/server', () => ({
+    NextResponse: {
+        next: vi.fn(() => 'NextResponse.next() called'),
+        json: vi.fn((body: any, init?: any) => ({ body, ...init })),
+    },
+}))
+
+describe('middleware', () => {
+    const [AUTH_USER, AUTH_PASS] = (process.env.HTTP_BASIC_AUTH || 'admin:password').split(':')
+
+    beforeEach(() => {
+        vi.resetAllMocks()
+    })
+
+    const createMockRequest = (authHeader?: string): Partial<NextRequest> => ({
+        headers: {
+            // @ts-ignore
+            get: vi.fn((key: string) => {
+                return key.toLowerCase() === 'authorization' ? authHeader : null
```

Comment:
Do the tests cause different casings of the string `authorization` to be passed in? Could the code be clean up to be consistent instead?